src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -45,20 +47,56 @@ public static string HtmlEncode(string value)
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlEncode(value, index, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            var sb = new PooledStringBuilder(value.Length);
+            try
+            {
+                HtmlEncode(value, index, ref sb);
+                return sb.GetStringAndClear();
+            }
+            finally
+            {
+                sb.Clear();
+            }
         }
 
         public static void HtmlEncode(string value, TextWriter output)
         {
-            output.Write(HtmlEncode(value));
+            if (output == null)
+            {
+                throw new ArgumentNullException(nameof(output));
+            }
+            if (string.IsNullOrEmpty(value))
+            {
+                output.Write(value);
+                return;
+            }
+
+            // Don't create StringBuilder if we don't have anything to encode
+            int index = IndexOfHtmlEncodingChars(value, 0);
+            if (index == -1)
+            {
+                output.Write(value);
+                return;
+            }
```

Comment:
How valuable is this check? In cases where we do have something to encode, we end up paying for the extra checking.

Reply:
I just copied this from the existing `string` overload of `HtmlEncode` above. My gut tells me that the case when `value` doesn't have any encoding characters is quite common, probably more common than when it does. Plus `index` is used to copy the prefix of the string quickly, see [line 106](https://github.com/dotnet/corefx/pull/27250/files/67f6909e4581843ecfa2b5ab23050a17edaeb4d9#diff-831332bbb389db07350457a4d92095f8R106)